### PR TITLE
Editorial: Update explanation for new spec structure

### LINKED ISSUE #91: Specification talks about non-existant [[EvaluationResult]] field

### PR DESCRIPTION
**Pull request by littledan**
Fixes #91

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -687,7 +687,7 @@ <h1>Example Source Text Module Record Graphs</h1>
...
+    <p>Finally, consider a case involving evaluation errors. If InnerModuleEvaluation of _C_ succeeds but, thereafter, fails for _B_, for example because _B_ contains code that throws an exception, then the original _A_.Evaluate() will fail<ins>, returning a Promise resolving to *undefined*</ins>. The resulting exception will be recorded in both _A_ and _B_'s [[EvaluationError]] fields, and their [[Status]] will become `"evaluated"`. _C_ will also become `"evaluated"` but, in contrast to _A_ and _B_, will remain without an [[EvaluationError]], as it successfully completed evaluation. Storing the exception ensures that any time a host tries to reuse _A_ or _B_ by calling their Evaluate() method, it will encounter the same exception. (Hosts are not required to reuse Source Text Module Records; similarly, hosts are not required to expose the exception objects thrown by these  methods. However, the specification enables such uses.)</p>
```
Comment (guybedford) [inline spec.html]: Should we remove the Promise note here too?

File: spec.html [hunk 2]
```diff
@@ -711,7 +711,7 @@ <h1>Example Source Text Module Record Graphs</h1>
...
+    <p>Finally, consider a case where _A_ has an evaluation error; for example, its source code throws an exception. In that case, the evaluation-time analog of the above steps still occurs, including the early return from the second call to InnerModuleEvaluation on _A_. However, once we unwind back to the original InnerModuleEvaluation on _A_, it fails by assumption. The exception thrown propagates up to _A_.Evaluate(), which records the error in all modules that are currently on its _stack_ (i.e., the modules that are still `"evaluating"`) <ins>as well as via [[AsyncParentModules]], which form a chain through the whole dependency graph through the CyclicModuleExecutionRejected algorithm</ins>. Hence both _A_ and _B_ become `"evaluated"` and the exception is recorded in both _A_ and _B_'s [[EvaluationError]] fields, while _C_ is left as `"evaluated"` with no [[EvaluationError]].</p>
```
Comment (guybedford) [inline spec.html]: Should we perhaps clarify on this line that `AsyncParentModules` only tracks dependency chain parents for async modules?

File: spec.html (hunk 1 above)
Comment (littledan) [reply spec.html]: Won't it return a rejected Promise?

File: spec.html (hunk 2 above)
Comment (littledan) [reply spec.html]: Added

File: spec.html (hunk 1 above)
Comment (guybedford) [reply spec.html]: Ahh, yes, thought it was referring to `InnerModuleEvaluation`.

**Pull request merged**